### PR TITLE
Sscs 3241 timeout warning

### PR DESCRIPTION
**Pull request by barbaracassani**
To test this, change the `inactivityAlert` in config/default.json . It's currently set to 15 minutes.

I tried to write tests with jsdom and then with cheerio, but somehow the plugin won't comply. Will investigate a bit further, but the substance of this pr is done.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -137,6 +137,13 @@ lookAndFeel.configure(app, {
       isArray(value) {
         return Array.isArray(value);
       },
+      inactivityTimeout: {
+        title: `${content.inactivityTimeout.title}`,
```
Comment (bendiggle) [inline app.js]: These don't need to be template literals `${content.inactivityTimeout.title}` can just be `content.inactivityTimeout.title`

File: assets/js/inactivity-alert.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  destroy() {
+    window.clearTimeout(this.timeoutForModal);
+    window.clearTimeout(this.timeoutForSession);
+    this.elExtend.off('click');
+    this.elDestroy.off('click');
+  }
+}
+
+
+export default InactivityAlert;
```
Comment (bendiggle) [inline assets/js/inactivity-alert.js]: need new line here

File: assets/js/main.js [hunk 3]
```diff
@@ -24,8 +32,47 @@ function initDatePicker() {
...
   initShowHideContent();
   initAutocomplete();
+  initSYAAnalyticsTrack();
+  initTM(redis.timeout, frontend.inactivityAlert);
   initDatePicker();
 });
+
+$(window).on('unload', () => {
+  destroyTM();
+});
```
Comment (bendiggle) [inline assets/js/main.js]: need new line here

File: views/components/form-template.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+</noscript>
+{% endblock %}
+
+
+{% block full_width %}
+{{ super() }}
+<div>
+    {% include "../components/inactivity-alert.html" %}
+</div>
+{% endblock %}
```
Comment (bendiggle) [inline views/components/form-template.html]: need new line here

File: views/components/inactivity-alert.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        <p>
+            {{ inactivityTimeout.expiringIn }}
+        </p>
+        <p>{{ inactivityTimeout.text }}</p>
+        <p>
+            <button class="button secondary extend" id="extend" href="#" rel="modal:close">{{ inactivityTimeout.yes }}</button>
+            <button class="button secondary abort" id="destroy" href="#">{{ inactivityTimeout.no }}</button>
+        </p>
+    </div>
+</div>
```
Comment (bendiggle) [inline views/components/inactivity-alert.html]: need new line here

File: config/default.json [hunk 6]
```diff
@@ -10,6 +10,9 @@
     "useSSL": "false",
     "timeout": 1200
   },
+  "frontend": {
+    "inactivityAlert": 900
```
Comment (bendiggle) [inline config/default.json]: will need to add this to `custom-environment-variables.json` as well.

Comment (barbaracassani) [reply config/default.json]: done 👍

File: assets/js/inactivity-alert.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+      .setTimeout(InactivityAlert.navigateAway, this.sessionSeconds * secondsToMilliseconds);
+  }
+  setTimeoutForModal() {
+    const el = $('#timeout-dialog');
+    this.timeoutForModal = window
+      .setTimeout(() => {
+        const count = 1000;
+        /* eslint-disable no-magic-numbers */
+        let startTime = 120000;
+        /* eslint-enable no-magic-numbers */
```
Comment (barbaracassani) [inline assets/js/inactivity-alert.js]: Silly eslint -- doesn't see it's reassigned later.

File: assets/js/main.js [hunk 8]
```diff
@@ -24,15 +32,58 @@ function initDatePicker() {
   }
 }
 
+function initSYAAnalyticsTrack() {
```
Comment (paulgain) [inline assets/js/main.js]: There is no need to add Google Analytics (GA) here, we already have this set up via Google Tag Manager (GTM).

File: assets/js/main.js [hunk 9]
```diff
@@ -24,15 +32,58 @@ function initDatePicker() {
...
+
+  // Set custom dimensions before tracking pageviews
+  // analytics.setDimension(…)
+
+  // Activate any event plugins eg. print intent, error tracking
+  // analyticsPlugins.error();
+  // analyticsPlugins.printIntent();
+
+  // Track initial pageview
+  window.GOVUK.analytics.trackPageview();
```
Comment (paulgain) [inline assets/js/main.js]: We are already tracking page views via GTM, there is no need to track each page twice!

**Pull request merged**